src/app/file-manager/file-modal/file-modal.component.ts

```diff
@@ -1,43 +1,104 @@
-import {Component, Inject} from '@angular/core';
+import {Component, Inject, Output, EventEmitter} from '@angular/core';
 import {MatDialog, MatDialogRef, MAT_DIALOG_DATA} from '@angular/material';
+import {TreeNode} from 'primeng/primeng';
+import { Item } from '@app/file-manager/models/item';
+import * as Downloadables from '../actions/downloadables.actions';
+import { Store } from '@ngrx/store';
+import { AppState } from '@app/file-manager/transferables-grid/transferables-grid.component';
+import { TransferablesGridComponent } from '../transferables-grid/transferables-grid.component';
+import { Message } from 'primeng/components/common/api';
+import { Status } from '@app/file-manager/models/status';
 
 @Component({
   selector: 'app-file-modal',
   templateUrl: './file-modal.component.html',
   styleUrls: ['./file-modal.component.scss']
 })
 export class FileModalComponent  {
+  @Output('done') done: EventEmitter<any> = new EventEmitter();
 
   preserveStructure = true;
   directory = 'Choose Directory...';
   isValid = false;
+  dataFile: Item;
+  files: TreeNode[];
+  selectedFiles: TreeNode[] = [];
+  selectedFile: TreeNode;
+  msgs: Message[] = [];
 
   constructor(
+    private transferablesGridComponent: TransferablesGridComponent,
+    private store: Store<AppState>,
     public dialogRef: MatDialogRef<FileModalComponent>,
     @Inject(MAT_DIALOG_DATA) public data: any) {
       if (localStorage.getItem('directory') !== null) {
         this.directory = localStorage.getItem('directory');
         this.isValid = true;
-      }
-     }
-
-    startDownload(): void {
-      this.dialogRef.close({directory: this.directory, preserveStructure: this.preserveStructure});
     }
+  }
 
-    cancel(): void {
-      this.dialogRef.close();
-    }
+  startDownload(): void {
+    this.setItems(Status.VERIFYING);
+    this.transferablesGridComponent.startDownload().then(
+      status => {
+        this.msgs = [];
+        if (!status.diskSpace) {
+          this.createWarningMsg('Sorry, there is not enough space on disk.');
+        } else if (!status.allowed) {
+          this.createWarningMsg('Sorry, you don\'t have the proper permissions to access that location.');
+        } else {
+          this.store.dispatch(new Downloadables.RemoveByStatus(Status.VERIFYING));
+          this.setItems(Status.DOWNLOADING);
```

Comment:
Instead of doing it this way, could you change the state of the items in the list from `VERIFYING` to `DOWNLOADING`? That seems like a better way to model the state of the transfer items.

Reply:
This has been changed, so now there is no necessity to handle this status from the store.